deployment/postgres-init-db/sql/tables/ui/seen_sources.sql

```diff
@@ -0,0 +1,28 @@
+-- Create a table to track which sources the user has and has not seen added/removed
+create table ui.seen_sources
+(
+    username text not null,
+    external_source_name text not null,
+    external_source_type text not null, -- included for ease of filtering, though not part of pkey
```

Comment:
Could you explain the filtering being done here? To be clear, I'm not against the column, I just want to understand how it's used.

Additionally, if this column is being commonly used as a filter, you should add an [index](https://www.postgresql.org/docs/current/sql-createindex.html) on it, similar to how we have an index on `plan_id` on the `activity_directive` table.

Reply:
The filtering is in the UI; we just include it so we don't have to do any more table crossreferencing/pulling in the frontend to get the source type.

Reply:
Seen sources has been largely refactored; as such this comment is now not applicable.